buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -87,16 +87,30 @@ class PrecommitTasks {
     }
 
     private static Task configureCheckstyle(Project project) {
+        // Always copy the checkstyle configuration files to 'buildDir/checkstyle' since the resources could be located in a jar
+        // file. If the resources are located in a jar, Gradle will fail when it tries to turn the URL into a file
+        File checkStyleDir = new File(project.buildDir, "checkstyle")
+        Task copyCheckstyleConf = project.tasks.create("copyCheckstyleConf")
+        copyCheckstyleConf.doLast {
+            checkStyleDir.mkdirs()
+            // withStream will close the output stream and #getBytes reads the InputStream fully and closes it
+            new FileOutputStream(new File(checkStyleDir, "checkstyle.xml")).withStream {
+                it.write(PrecommitTasks.getResource("/checkstyle.xml").openStream().getBytes())
+            }
+            new FileOutputStream(new File(checkStyleDir, "checkstyle_suppressions.xml")).withStream {
+                it.write(PrecommitTasks.getResource("/checkstyle_suppressions.xml").openStream().getBytes())
+            }
+        }
+
         Task checkstyleTask = project.tasks.create('checkstyle')
         // Apply the checkstyle plugin to create `checkstyleMain` and `checkstyleTest`. It only
         // creates them if there is main or test code to check and it makes `check` depend
         // on them. But we want `precommit` to depend on `checkstyle` which depends on them so
         // we have to swap them.
         project.pluginManager.apply('checkstyle')
-        URL checkstyleSuppressions = PrecommitTasks.getResource('/checkstyle_suppressions.xml')
+        File checkstyleSuppressions = new File(checkStyleDir, "checkstyle_suppressions.xml")
         project.checkstyle {
-            config = project.resources.text.fromFile(
-                PrecommitTasks.getResource('/checkstyle.xml'), 'UTF-8')
+            config = project.resources.text.fromFile(new File(checkStyleDir, "checkstyle.xml"), 'UTF-8')
```

Comment:
instead of creating a file copying the content etc, you can probably make this a single line change:
`config = project.resources.text.fromString(new String(PrecommitTasks.getResource("/checkstyle_suppressions.xml").openStream().getBytes()))`